pkgs/README.md

```diff
@@ -327,43 +327,53 @@ A (typically large) program with a distinct user interface, primarily used inter
+- The variable name used for the instantiated package in [by-name structure](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md) or `all-packages.nix`, and when passing it as a dependency to other functions. Typically this is called the _package attribute name_. This is what Nix expression authors see. It can also be used when installing using `nix-env -iA`.
 
 - The filename for (the directory containing) the Nix expression.
 
 Most of the time, these are the same. For instance, the package `e2fsprogs` has a `pname` attribute `"e2fsprogs"`, is bound to the variable name `e2fsprogs` in `all-packages.nix`, and the Nix expression is in `pkgs/os-specific/linux/e2fsprogs/default.nix`.
 
-There are a few naming guidelines:
+These are a few naming guidelines:
 
-- The `pname` attribute _should_ be identical to the upstream package name.
+- For `pname` attribute:
 
-- The `pname` and the `version` attribute _must not_ contain uppercase letters — e.g., `"mplayer"` instead of `"MPlayer"`.
+  - It _should_ be identical to the upstream package name.
 
-- The `version` attribute _must_ start with a digit e.g., `"0.3.1rc2"` or `"0-unstable-1970-01-01"`.
+  - It _must not_ contain uppercase letters — e.g., `"mplayer"` instead of `"MPlayer"`.
 
-- If a package is a commit from a repository without a version assigned, then the `version` attribute _should_ be the latest upstream version preceding that commit, followed by `-unstable-` and the date of the (fetched) commit. The date _must_ be in `"YYYY-MM-DD"` format.
-
-Example: Given a project had its latest releases `2.2` in November 2021, and `3.0` in January 2022, a commit authored on March 15, 2022 for an upcoming bugfix release `2.2.1` would have `version = "2.2-unstable-2022-03-15"`.
+- For variable name of the package:
 
-- If a project has no suitable preceding releases - e.g., no versions at all, or an incompatible versioning / tagging schema - then the latest upstream version in the above schema should be `0`.
+  - It _must_ be a valid variable name in Nix.
 
-Example: Given a project that has no tags / released versions at all, or applies versionless tags like `latest` or `YYYY-MM-DD-Build`, a commit authored on March 15, 2022 would have `version = "0-unstable-2022-03-15"`.
+  - If the `pname` starts with a digit, the variable name _should_ be prefixed with an underscore — e.g., the corresponding variable name for `0ad` should be `_0ad`. Otherwise the variable name _should not_ be prefixed with an underline.
 
-- Dashes in the package `pname` _should_ be preserved in new variable names, rather than converted to underscores or camel cased — e.g., `http-parser` instead of `http_parser` or `httpParser`. The hyphenated style is preferred in all three package names.
+  - Dashes in the package `pname` _should_ be preserved in new variable names, rather than converted to underscores or camel cased — e.g., `http-parser` instead of `http_parser` or `httpParser`.
```

Comment:
Does this infer that hyphenate is more preferable than snake/camel case? Should we convert snake/camel case to hyphenate?

Reply:
> Does this infer that hyphenate is more preferable than snake/camel case?

I read this as that we should keep whatever is in `pname`, but we could be more concrete about this.

```suggestion
  - New attribute names _should_ be the same as the value in `pname`.

     Do not convert hyphenated names to to [snake case](https://en.wikipedia.org/wiki/Snake_case) or [camel case](https://en.wikipedia.org/wiki/Camel_case).
      This was done historically, but is not necessary any more.
      [The Nix language allows dashes in identifiers since 2012](https://github.com/NixOS/nix/commit/95c74eae269b2b9e4bc514581b5caa1d80b54acc).
     
     Example: Use `http-parser` instead of `http_parser` or `httpParser`.
```